jwst/wfs_combine/wfs_combine.py

```diff
@@ -81,21 +86,42 @@ def do_all(self):
             combined input file data
         """
 
+        self.off_x, self.off_y = self.get_wcs_offsets()
+
+        # If the shift in x is negative, switch the two images
+        if self.off_x < 0 and self.flip_dithers:
+            self.input_1.close()
+            self.input_2.close()
+            self.input_1 = datamodels.open(self.file2)
+            self.input_2 = datamodels.open(self.file1)
+            log.info('File 1 to combine: %s', self.file2)
+            log.info('File 2 to combine: %s', self.file1)
+        else:
+            log.info('File 1 to combine: %s', self.file1)
+            log.info('File 2 to combine: %s', self.file2)
```

Comment:
Having files opened in `__init__()` and closed in `do_all()` is not ideal.  File management should probably be happening all in `__init__()`.  Perhaps this should go in `__init__()` for the class?

Reply:
Or better yet, define `self.__enter__` and `self.__exit__` methods for the class, so that the `with` context manager can be used.

Reply:
I was trying to fit within the framework of the original code. Having a single-use class for this step doesn't make much sense to me and seems to be used to just have global variables (class attributes) to minimize parameter passing.
The close within the if statement is because we need to switch the contents of input_1 and input 2.  This is only called when the images need to be flipped. The files are left open in both cases of the if and are closed when the step finishes.  If I move the open to __enter__ and close to __exit__, I still have to close it and reopen it when the images need to be flipped.